src/interceptor.ts

```diff
@@ -1,12 +1,20 @@
-import { AxiosRequestConfig, AxiosRequestHeaders, Method } from "axios";
-import { sign } from "aws4";
+import { Request as AWS4Request, sign } from "aws4";
+import {
+  AxiosHeaders,
+  AxiosRequestConfig,
+  AxiosRequestHeaders,
+  InternalAxiosRequestConfig,
+  Method,
+} from "axios";
 import buildUrl from "axios/lib/helpers/buildURL";
 import combineURLs from "axios/lib/helpers/combineURLs";
 import isAbsoluteURL from "axios/lib/helpers/isAbsoluteURL";
```

Comment:
The getUri can replace the need for the helpers functions
```suggestion
  AxiosRequestHeaders,
  // @ts-ignore
  getUri,
  InternalAxiosRequestConfig,
  Method,
} from "axios";
```

Reply:
Then line 107-116 can be replaced with:
```typescript
    const url = getUri(config);

    if (config.params) {
      delete config.params;
    }
```